### PR TITLE
Eliminate FLUTTER_NOLINT where possible

### PR DESCRIPTION
**Pull request by cbracken**
## Description

This removes most of the remaining FLUTTER_NOLINT comments and opts
these files back into linter enforcement.

I've filed https://github.com/flutter/flutter/issues/68273 to require
that all FLUTTER_NOLINT comments be followed by a GitHub issue URL
describing the problem to be fixed.

## Related Issues

https://github.com/flutter/flutter/issues/68273: Update engine linter to require a bug URL on FLUTTER_NOLINT lines

## Tests

No tests required. The linter will do the 'testing'.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [ ] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: shell/platform/embedder/embedder.cc [hunk 1]
```diff
@@ -1,8 +1,6 @@
 // Copyright 2013 The Flutter Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
-// FLUTTER_NOLINT
-// FLUTTER_NOLINT
```
Comment (cbracken) [inline shell/platform/embedder/embedder.cc:5]: I'm going to take this as a double-negative. :heavy_check_mark:

**Comment by cbracken** [comment]
Cirrus failure was an unrelated lint issue in the framework. This was fixed by flutter/flutter#68136.

**Review by stuartmorgan — APPROVED** [review]
Weird, I thought these had only been added to files that didn't pass linting when it was turned on?

LGTM since it's passing bots though!

**Comment by cbracken** [comment]
Landing on purple infra flake in hopes of unblocking head.

**Comment by chinmaygarde** [comment]
The infra flake is being [tracked here](https://github.com/flutter/flutter/issues/68328). The build that this patch triggered also failed because of the same reason.

**Pull request merged**